### PR TITLE
Mark the AdMob SDK as deprecated

### PR DESCRIPTION
**Pull request by DellaBitta**
- Mark AdMob SDK as deprecated.
- Merge to main upon approval of feature/admob_2021 PR.

***

- Local test on iOS and Android.
- [Integration Test](https://github.com/firebase/firebase-cpp-sdk/actions/runs/2314502065).

***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [X] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Comment by jonsimantov** [comment]
This doesn't go in the admob_2021 branch?

**Comment by DellaBitta** [comment]
We discussed at the weekly and resolved on having it go directly into the main branch as to not clutter up the feature branch.

**Comment by DellaBitta** [comment]
But I can point it to the feature branch if you'd like.

File: admob/src/include/firebase/admob/types.h [hunk 1]
```diff
@@ -159,8 +225,16 @@ struct AdRequest {
   ChildDirectedTreatmentState tagged_for_child_directed_treatment;
 };
 
+/// @deprecated
```
Comment (jonsimantov) [inline admob/src/include/firebase/admob/types.h:228]: Does our docs generation turn text after the \@deprecated into a note in the docs itself? If so, should we put something here?

Comment (DellaBitta) [reply admob/src/include/firebase/admob/types.h:228]: I tried that at first and the formatting was pretty horrible. The deprecated message appeared before all of the @brief text, so it didn't read well.  I have a screen shot I'll upload tomorrow.

**Review by jonsimantov — APPROVED** [review]

File: admob/src/include/firebase/admob/types.h (hunk 1 above)
Comment (DellaBitta) [reply admob/src/include/firebase/admob/types.h:228]: This is what the brief summary renders as if I create a message after the @deprecation tag:

```
/// @brief Iniitalizes AdMob via Firebase
///
/// <detailed description>
///
/// @deprecated <deprecation message>
Future<AdapterInitializationStatus> Initialize(const ::firebase::App& app, InitResult* init_result_out = nullptr);
```
Screen Shot 2022-05-10 at 8 34 58 AM

I personally prefer this format better.

Comment (jonsimantov) [reply admob/src/include/firebase/admob/types.h:228]: Hmm, is it able to link to firebase::gma namespace? (We might need to make some placeholder content like we did with Future on the gma side...)

**Pull request merged**